### PR TITLE
ROC-5749: Staff email when claimant intends to proceed with directions hearing

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[ROC-5749](https://tools.hmcts.net/jira/browse/ROC-5749)

### Change description ###
This PR implements the notification to staff when claimant intends to proceed with directions hearing.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/ClaimantRejectionStaffNotificationService.java [hunk 1]
```diff
@@ -80,7 +108,6 @@ public void notifyStaffClaimantRejectPartAdmission(Claim claim) {
 
     private EmailAttachment createResponsePdfAttachment(Claim claim) {
         requireNonNull(claim);
-        System.out.println(claim);
```
Comment (markry-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/ClaimantRejectionStaffNotificationService.java]: 👀

File: src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/emails/StaffEmailTemplates.java [hunk 2]
```diff
@@ -102,4 +102,12 @@ public String getClaimantRejectPartAdmissionEmailBody() {
     public String getClaimantRejectPartAdmissionEmailSubject() {
         return readString("/staff/templates/email/claimantResponse/rejection/partAdmission/subject.txt");
     }
+
+    public String getClaimantDirectionsHearingEmailSubject() {
+        return readString("/staff/templates/email/claimantResponse/rejection/directionsHearing/subject.txt");
```
Comment (markry-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/emails/StaffEmailTemplates.java]: txt files missing from this PR

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/ClaimantRejectionStaffNotificationService.java [hunk 3]
```diff
@@ -58,6 +63,29 @@ public void notifyStaffClaimantRejectPartAdmission(Claim claim) {
...
+                .to(staffEmailProperties.getRecipient())
+                .subject(emailContent.getSubject())
+                .message(emailContent.getBody())
+                .build()
+        );
+    }
+
+    public static Map<String, Object> getParameters(Claim claim) {
+        return new ImmutableMap.Builder<String, Object>()
+            .put("claimReferenceNumber", claim.getReferenceNumber())
```
Comment (dharmendrak) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/ClaimantRejectionStaffNotificationService.java]: can we use constant already defined for these keys

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/ClaimantRejectionStaffNotificationService.java]: done

File: src/main/resources/staff/templates/email/claimantResponse/rejection/directionsHearing/body.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+The claimant has reviewed the defendant’s response and wish to proceed with the claim if they opted into mediation the MILO team has been sent their contact & claim details.
```
Comment (andrewwa-kainos) [inline src/main/resources/staff/templates/email/claimantResponse/rejection/directionsHearing/body.txt]: Was this content reviewed? Could use some punctuation and grammatical changes

File: src/test/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    public void shouldNotNotifyDefendantOfIntentToProceedWhenClaimantDoesNotIntentToProceed() {
+        Claim claim = SampleClaim.builder()
+            .withClaimantRespondedAt(LocalDateTime.now())
+            .withClaimantResponse(SampleClaimantResponse.validDefaultRejection())
+            .build();
+        ClaimantResponseEvent event = new ClaimantResponseEvent(claim);
+        handler.notifyDefendantOfIntentToProceed(event);
+
+        verify(notificationService, times(0))
+            .notifyDefendantOfClaimantResponse(eq(event.getClaim()));
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionHandlerTest.java]: `once()` is redundant
`times(0)` should be `never()`
`eq()` isn't needed in this case

File: src/main/resources/staff/templates/email/claimantResponse/rejection/directionsHearing/body.txt (hunk 4 above)
Comment (sabahirfan) [reply src/main/resources/staff/templates/email/claimantResponse/rejection/directionsHearing/body.txt]: yes reviewed by PO.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionHandlerTest.java (hunk 5 above)
Comment (sabahirfan) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionHandlerTest.java]: done. 👍 

`never()` is a wrapper around `times(0)`. 
```
 @CheckReturnValue
    public static VerificationMode never() {
        return times(0);
    }
```

**Pull request merged**